Add Niadd extension for Portuguese source

Zev-Kaapor opened the pull request:
Introduces a new extension for the Niadd manga source in Portuguese, including the build configuration, launcher icons for various resolutions, and the main source implementation with support for popular, latest, search, manga details, chapters, and page listing.

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

closes: #3886

Commit:
Add Niadd extension for Portuguese source

Introduces a new extension for the Niadd manga source in Portuguese, including the build configuration, launcher icons for various resolutions, and the main source implementation with support for popular, latest, search, manga details, chapters, and page listing.

AwkwardPeak7 reviewed (commented):
Niadd supports more than Portuguese, this need to be a source factory with mutiple languages, take a look at something like [Comick extension](https://github.com/keiyoushi/extensions-source/tree/d88c256a8bc1dabf7f27ea1827efa69e71700116/src/all/comicklive) to see how source factory works.

Commit:
Add Niadd multi-language extension

Introduces a new extension for Niadd supporting multiple languages (pt-BR, en, es, it, ru, de, fr). Includes source implementation, factory, build configuration, and launcher icons for various resolutions.

Zev-Lonewolf commented:
That's right, I redesigned the extension to support multiple languages and fixed a small redirection error in more recent works that use a certain “Select source” feature, which is nothing more than a well-disguised ad.

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,260 @@
+    langParam: String,
+) : ParsedHttpSource() {
+
+    override val name = "Niadd$nameSuffix"
```
Remove suffix. The app handles multiple extensions with the same name.
```suggestion
    override val name = "Niadd"
```

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,260 @@
+
+    override fun popularMangaFromElement(element: Element): SManga =
+        SManga.create().apply {
+            title = element.select("div.manga-name").text()
```
```suggestion
            title = element.selectFirst("div.manga-name")!!.text()
```

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,260 @@
+        SManga.create().apply {
+            title = element.select("div.manga-name").text()
+            val rawUrl = element.select("a").first()?.attr("href").orEmpty()
+            url = rawUrl.removePrefix(baseUrl)
```
```suggestion
            setUrlWithoutDomain(rawUrl)
```

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,260 @@
+            if (match != null) {
+                val content = match.groupValues[1]
+                val urls = content.split(",")
+                    .map { it.replace(Regex("""["'\s]"""), "").trim() }
```
Move static regex to class.

Commit:
Remove Niadd Portuguese extension and update all/niadd

Deleted the Niadd Portuguese extension, including its source code, resources, and build files. Refactored the all/niadd extension to consolidate logic, improve code quality, and ensure continued support for Portuguese within the unified extension.

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,258 @@
+    // Details
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        return GET(baseUrl + manga.url, headers)
+    }
```
Useless override, can be removed.

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,258 @@
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        val infoElement = document.select("div.bookside-general, div.detail-general")
+
+        title = document.selectFirst("h1, .book-headline-name")?.text().orEmpty()
```
Title is required
```suggestion
        title = document.selectFirst("h1, .book-headline-name")!!.text()
```

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,258 @@
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val chapterListContainer = document.select("ul.chapter-list")
```
```suggestion
        val chapterListContainer = document.selectFirst("ul.chapter-list")!!
```

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,258 @@
+        val chapterListContainer = document.select("ul.chapter-list")
+
+        chapterListContainer.removeClass("dis-hide")
+        document.select(".chp-warn-box").remove()
```
Do you need to modify the document? It seems to have no effect.

vetleledaal commented on src/all/niadd/src/eu/kanade/tachiyomi/extension/all/niadd/Niadd.kt:
```diff
@@ -0,0 +1,258 @@
+    // Pages
+    override fun chapterFromElement(element: Element): SChapter =
+        SChapter.create().apply {
+            val rawUrl = element.attr("href")
```
```suggestion
            val rawUrl = element.absUrl("href")
```
```suggestion
            val rawUrl = element.attr("abs:href")
```

\*Jsoup has additional context needed to resolve some relative paths, `setUrlWithoutDomain` does not.

Commit:
Refactor Niadd extension for improved parsing

Refactored search request to use HttpUrl builder for safer query parameter handling. Improved null safety and parsing logic in manga details and chapter list parsing, including stricter element selection and more robust date parsing. Cleaned up unused code and enhanced reliability of chapter URL and name extraction.

Commit:
Remove custom User-Agent header from Niadd extension

The explicit User-Agent header was removed from the headersBuilder in Niadd.kt, reverting to the default behavior. This may improve compatibility or reduce maintenance overhead.

Commit:
Set User-Agent header to bypass Niadd hotlink protection

Added a hardcoded User-Agent in headersBuilder to bypass Niadd's hotlink protection for cover images. Also updated exception messages and set relative date parsing to return 0L as a placeholder.
